pkg/grizzly/providers.go

```diff
@@ -92,6 +98,21 @@ func (r *Resource) Spec() map[string]interface{} {
 	return (*r)["spec"].(map[string]interface{})
 }
 
+func (r *Resource) SpecToCheck() (synthetic_monitoring.Check, error) {
+	var smSpec SMSpec
+	data, err := json.Marshal((*r)["spec"])
+	if err != nil {
+		return synthetic_monitoring.Check{}, nil
+	}
+
+	err = json.Unmarshal(data, &smSpec)
+	if err != nil {
+		return synthetic_monitoring.Check{}, nil
+	}
+
+	return smSpec.Check, nil
+}
+
```

Comment:
does this even work? Really, this functionality needs to be in `grafana/synthetic-monitoring.go`, rather than having handler specific stuff in the generic package.